src/Workspaces/Core/Portable/Utilities/NoOpDisposable.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+
+namespace Roslyn.Utilities
+{
+    /// <summary>
+    /// Dummy no-op disposable.
+    /// </summary>
+    internal struct NoOpDisposable : IDisposable
```

Comment:
❕ This is unnecessary. Just change the `addOperationScope` method to have the signature `Func<string, IDisposable?>` and use a lambda that returns `null`.

Reply:
Yes, much better. Thanks!